app/api/users.js

```diff
@@ -0,0 +1,83 @@
+// User creation and authentication with password hashing.
+// Code sourced from Zybooks 11.12 https://learn.zybooks.com/zybook/UMASSCOMPSCI326RichardsAcademicYear2020/chapter/11/section/12
+const jwt = require("jwt-simple");
+const User = require("../models/user");
+const router = require("express").Router();
+const bcrypt = require("bcrypt-nodejs");
+const saltRounds = 10;
+
+router.post("/user", (req, res) => {
+    bcrypt.genSalt(saltRounds, (err, salt) => {
+        bcrypt.hash(req.body.password, salt, (err, hash) => {
```

Comment:
This is erroring. It looks like `bcrypt.hash` requires you to specify all 4 parameters (but the `progress` parameter can be `undefined`)